<?xml version="1.0"?>
<launch>
    <!--Number of corners-->
    <arg name="board_width"   default="8"/>
    <arg name="board_height"  default="5"/>

    <arg name="square_length" default="39"/><!--mm-->

    <arg name="image_width"  default="4000"/>
    <arg name="image_height" default="3000"/>

    <!--Calibration modes-->
    <!--0 Checker-->
    <!--1 Charuco-->
    <arg name="method" default="1"/>

    <!--Charuco marker length if required-->
    <arg name="marker_length" default="20.0"/><!--mm-->

    <node name="stereo_calibration_service" pkg="stereo_calibration" type="stereo_calibration_service" output="screen">
        <param name ="board_width"  value="$(arg board_width)"/>
        <param name ="board_height" value="$(arg board_height)"/>
        <param name ="square_length" value="$(arg square_length)"/>

        <param name ="image_width"  value="$(arg image_width)"/>
        <param name ="image_height" value="$(arg image_height)"/>

        <param name ="method" value="$(arg method)"/>

        <param name ="marker_length" value="$(arg marker_length)"/>
    </node>
</launch>
